### PR TITLE
Update for opening nominations

### PR DESCRIPTION
**Pull request by pavithraes**
The following are updated:

* Board responsibilities
* Election process - Soliciting nominations
* Timeline

### CODE REVIEW

File: election-process.md [hunk 1]
```diff
@@ -26,23 +26,17 @@ These messages will be published by NumFOCUS in the following spaces:
...
-This explicitly includes (but not limited to) people who associate with:
+We welcome and encourage participation from people of all backgrounds, identities, and geographic diversity.
 
-- Software development
-- Community management
-- Technical writing
-- Science and research
-- Legal industry
-- Financial industry
+Nominees for the **Technical Steering Committee** should be contributors or members of a NumFOCUS Sponsored or Affiliated Project community. They should have experience in areas such as governance or community leadership, budget creation and management, infrastructure, and strategic planning.
```
Comment (tkoyama010) [inline election-process.md:33]:
```suggestion
Nominees for the **Technical Steering Committee** should be contributors or members of a NumFOCUS Sponsored or Affiliated Project community.
They should have experience in areas such as governance or community leadership, budget creation and management, infrastructure, and strategic planning.
```

**Review by tkoyama010 — CHANGES_REQUESTED** [review]

**Comment by tkoyama010** [comment]
pre-commit.ci autofix

File: board-responsibilities.md [hunk 2]
```diff
@@ -1,26 +1,43 @@
 ## Responsibilities of Board of Directors and Technical Steering Committee
 
-> [!WARNING]
-> In development, more details will be added soon.
+The NumFOCUS Board of Directors and the Technical Steering Committee are positions of authority.
+The former group deals with higher-level strategic and administrative decisions, while the latter deals with more hands-on day-to-day technical decisions.
+
+This document broadly lists the responsibilities of each group.
 
 ## Board of Directors - Administrative or Executive Branch
```
Comment (lsilen) [inline board-responsibilities.md]: Delete Executive Branch and use Administrative Board to be consistent with what the board is using.

**Comment by pavithraes** [comment]
Thanks for the comments! This PR was also reviewed during the election committee meting, so I'll take the liberty to self-merge. :)

**Comment by tkoyama010** [comment]
@all-contributors please add @lsilen for review

**Pull request merged**